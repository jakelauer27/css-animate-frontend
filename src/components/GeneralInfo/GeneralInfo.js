import React, { Component } from 'react';

class GeneralInfo extends Component {
  
  render() {
    return (
      <div className="howto-info info-section">
        <h1 className='title'>CSS Ani<span>Mate</span></h1>
        <p>This is a place for you to learn and experiment with CSS animations.  
          Browse through a library of prebuilt animations and tweak them 
          however you please or login and create you own. Then, when you have adjusted the animation as you
          like, copy the code into your own project using the copy feature. 
        </p> 
        <p>
          If you arenâ€™t familiar with how to use keyframes or the animation 
          property, checkout the keyframes and animation property info 
          sections by clicking the tabs on the top of this popup.
          This app is meant to give an introduction to CSS animations.
          For more detailed info, please visit 
            <a className='link' target='_blank' rel="noopener noreferrer" href='https://www.w3schools.com/css/css3_animations.asp'> W3schools </a> or 
            <a className='link' target='_blank' rel="noopener noreferrer" href='https://developer.mozilla.org/en-US/docs/Web/CSS/animation'> MDN </a>.
        </p>
         <p className='howto-final-line'>Happy <span className='animating-span'>animating</span>!</p>
      </div>
    );
  }
}

export default GeneralInfo