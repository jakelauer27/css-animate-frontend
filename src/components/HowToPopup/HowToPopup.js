import React, { Component } from 'react'
import GeneralInfo from '../GeneralInfo/GeneralInfo'
import PropertiesInfo from '../PropertiesInfo/PropertiesInfo'
import KeyframesInfo from '../KeyframesInfo/KeyframesInfo'
import { NavLink, Route, Switch, Link } from 'react-router-dom'


export default class HowToPopup extends Component {

  render() {
    let { animation } = this.props
    if (!animation) {animation = 'slideInX'}
    return (
      <div className="infoPopup-overlay">
        <div className="info-popup">
          <header className="header">
              <NavLink 
                to={`/properties/howto/general`}
                activeClassName='selected-section'
                className='howTo header-button'>
                how to play
              </NavLink>
              <NavLink 
                to={`/properties/howto/properties`}
                activeClassName='selected-section'
                className='keyframes header-button'>
                keyframes
              </NavLink>
              <NavLink 
                to={`/properties/howto/keyframes`}
                activeClassName='selected-section'
                className='animation header-button'>
                animation properties
              </NavLink>
          </header>
          {
            <Switch>
              <Route path={`/properties/howto/general`}
                component={GeneralInfo} />
              <Route path={`/properties/howto/properties`}
                component={PropertiesInfo} />
              <Route path={`/properties/howto/keyframes`}
                component={KeyframesInfo} />
            </Switch>
          }
        <Link to={`/properties`} className='close-info-popup-btn'>
          Go!
        </Link>
        </div>
      </div>
    )
  }
}
