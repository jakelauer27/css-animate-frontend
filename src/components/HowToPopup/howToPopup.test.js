import PropertiesInfo from '../PropertiesInfo/PropertiesInfo'
import GeneralInfo from '../GeneralInfo/GeneralInfo'
import KeyframesInfo from '../KeyframesInfo/KeyframesInfo'
import React from 'react'
import { shallow, mount } from 'enzyme'
import HowToPopup from './HowToPopup'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'


describe('HowToPopup', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<HowToPopup />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('routes', () => {
    const mockStore = {getState: jest.fn(), subscribe: jest.fn(), dispatch: jest.fn()}

    it('should render GeneralInfo if on the GeneralInfo route', () => {
      let wrapper = mount(
        <Provider store={mockStore}>
          <MemoryRouter initialEntries={['/properties/howto/general']}>
            <HowToPopup></HowToPopup>
          </MemoryRouter>
        </Provider>
        )
  
        expect(wrapper.find(GeneralInfo)).toHaveLength(1)
    })

    it('should render PropertiesInfo if on the PropertiesInfo route', () => {
      let wrapper = mount(
        <Provider store={mockStore}>
          <MemoryRouter initialEntries={['/properties/howto/properties']}>
            <HowToPopup></HowToPopup>
          </MemoryRouter>
        </Provider>
        )
  
        expect(wrapper.find(PropertiesInfo)).toHaveLength(1)
    })

    it('should render keyframesInfo if on the keyframesInfo route', () => {
      let wrapper = mount(
        <Provider store={mockStore}>
          <MemoryRouter initialEntries={['/properties/howto/keyframes']}>
            <HowToPopup></HowToPopup>
          </MemoryRouter>
        </Provider>
        )
  
        expect(wrapper.find(KeyframesInfo)).toHaveLength(1)
    })
  })
})