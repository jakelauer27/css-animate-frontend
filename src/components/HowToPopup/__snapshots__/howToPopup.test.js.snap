// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToPopup should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HowToPopup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="info-popup"
      >
        <header
          className="header"
        >
          <NavLink
            activeClassName="selected-section"
            aria-current="page"
            className="howTo header-button"
            to="/slideInX/properties/howto/general"
          >
            how to play
          </NavLink>
          <NavLink
            activeClassName="selected-section"
            aria-current="page"
            className="keyframes header-button"
            to="/slideInX/properties/howto/properties"
          >
            keyframes
          </NavLink>
          <NavLink
            activeClassName="selected-section"
            aria-current="page"
            className="animation header-button"
            to="/slideInX/properties/howto/keyframes"
          >
            animation properties
          </NavLink>
        </header>
        <Switch>
          <Route
            component={[Function]}
            path="/slideInX/properties/howto/general"
          />
          <Route
            component={[Function]}
            path="/slideInX/properties/howto/properties"
          />
          <Route
            component={[Function]}
            path="/slideInX/properties/howto/keyframes"
          />
        </Switch>
        <Link
          className="close-info-popup-btn"
          replace={false}
          to="/slideInX/properties"
        >
          Go!
        </Link>
      </div>,
      "className": "infoPopup-overlay",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="header"
          >
            <NavLink
              activeClassName="selected-section"
              aria-current="page"
              className="howTo header-button"
              to="/slideInX/properties/howto/general"
            >
              how to play
            </NavLink>
            <NavLink
              activeClassName="selected-section"
              aria-current="page"
              className="keyframes header-button"
              to="/slideInX/properties/howto/properties"
            >
              keyframes
            </NavLink>
            <NavLink
              activeClassName="selected-section"
              aria-current="page"
              className="animation header-button"
              to="/slideInX/properties/howto/keyframes"
            >
              animation properties
            </NavLink>
          </header>,
          <Switch>
            <Route
              component={[Function]}
              path="/slideInX/properties/howto/general"
            />
            <Route
              component={[Function]}
              path="/slideInX/properties/howto/properties"
            />
            <Route
              component={[Function]}
              path="/slideInX/properties/howto/keyframes"
            />
          </Switch>,
          <Link
            className="close-info-popup-btn"
            replace={false}
            to="/slideInX/properties"
          >
            Go!
          </Link>,
        ],
        "className": "info-popup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="selected-section"
                aria-current="page"
                className="howTo header-button"
                to="/slideInX/properties/howto/general"
              >
                how to play
              </NavLink>,
              <NavLink
                activeClassName="selected-section"
                aria-current="page"
                className="keyframes header-button"
                to="/slideInX/properties/howto/properties"
              >
                keyframes
              </NavLink>,
              <NavLink
                activeClassName="selected-section"
                aria-current="page"
                className="animation header-button"
                to="/slideInX/properties/howto/keyframes"
              >
                animation properties
              </NavLink>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected-section",
                "aria-current": "page",
                "children": "how to play",
                "className": "howTo header-button",
                "to": "/slideInX/properties/howto/general",
              },
              "ref": null,
              "rendered": "how to play",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected-section",
                "aria-current": "page",
                "children": "keyframes",
                "className": "keyframes header-button",
                "to": "/slideInX/properties/howto/properties",
              },
              "ref": null,
              "rendered": "keyframes",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected-section",
                "aria-current": "page",
                "children": "animation properties",
                "className": "animation header-button",
                "to": "/slideInX/properties/howto/keyframes",
              },
              "ref": null,
              "rendered": "animation properties",
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/slideInX/properties/howto/general"
              />,
              <Route
                component={[Function]}
                path="/slideInX/properties/howto/properties"
              />,
              <Route
                component={[Function]}
                path="/slideInX/properties/howto/keyframes"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/slideInX/properties/howto/general",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/slideInX/properties/howto/properties",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/slideInX/properties/howto/keyframes",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Go!",
            "className": "close-info-popup-btn",
            "replace": false,
            "to": "/slideInX/properties",
          },
          "ref": null,
          "rendered": "Go!",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="info-popup"
        >
          <header
            className="header"
          >
            <NavLink
              activeClassName="selected-section"
              aria-current="page"
              className="howTo header-button"
              to="/slideInX/properties/howto/general"
            >
              how to play
            </NavLink>
            <NavLink
              activeClassName="selected-section"
              aria-current="page"
              className="keyframes header-button"
              to="/slideInX/properties/howto/properties"
            >
              keyframes
            </NavLink>
            <NavLink
              activeClassName="selected-section"
              aria-current="page"
              className="animation header-button"
              to="/slideInX/properties/howto/keyframes"
            >
              animation properties
            </NavLink>
          </header>
          <Switch>
            <Route
              component={[Function]}
              path="/slideInX/properties/howto/general"
            />
            <Route
              component={[Function]}
              path="/slideInX/properties/howto/properties"
            />
            <Route
              component={[Function]}
              path="/slideInX/properties/howto/keyframes"
            />
          </Switch>
          <Link
            className="close-info-popup-btn"
            replace={false}
            to="/slideInX/properties"
          >
            Go!
          </Link>
        </div>,
        "className": "infoPopup-overlay",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="header"
            >
              <NavLink
                activeClassName="selected-section"
                aria-current="page"
                className="howTo header-button"
                to="/slideInX/properties/howto/general"
              >
                how to play
              </NavLink>
              <NavLink
                activeClassName="selected-section"
                aria-current="page"
                className="keyframes header-button"
                to="/slideInX/properties/howto/properties"
              >
                keyframes
              </NavLink>
              <NavLink
                activeClassName="selected-section"
                aria-current="page"
                className="animation header-button"
                to="/slideInX/properties/howto/keyframes"
              >
                animation properties
              </NavLink>
            </header>,
            <Switch>
              <Route
                component={[Function]}
                path="/slideInX/properties/howto/general"
              />
              <Route
                component={[Function]}
                path="/slideInX/properties/howto/properties"
              />
              <Route
                component={[Function]}
                path="/slideInX/properties/howto/keyframes"
              />
            </Switch>,
            <Link
              className="close-info-popup-btn"
              replace={false}
              to="/slideInX/properties"
            >
              Go!
            </Link>,
          ],
          "className": "info-popup",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="selected-section"
                  aria-current="page"
                  className="howTo header-button"
                  to="/slideInX/properties/howto/general"
                >
                  how to play
                </NavLink>,
                <NavLink
                  activeClassName="selected-section"
                  aria-current="page"
                  className="keyframes header-button"
                  to="/slideInX/properties/howto/properties"
                >
                  keyframes
                </NavLink>,
                <NavLink
                  activeClassName="selected-section"
                  aria-current="page"
                  className="animation header-button"
                  to="/slideInX/properties/howto/keyframes"
                >
                  animation properties
                </NavLink>,
              ],
              "className": "header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "selected-section",
                  "aria-current": "page",
                  "children": "how to play",
                  "className": "howTo header-button",
                  "to": "/slideInX/properties/howto/general",
                },
                "ref": null,
                "rendered": "how to play",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "selected-section",
                  "aria-current": "page",
                  "children": "keyframes",
                  "className": "keyframes header-button",
                  "to": "/slideInX/properties/howto/properties",
                },
                "ref": null,
                "rendered": "keyframes",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "selected-section",
                  "aria-current": "page",
                  "children": "animation properties",
                  "className": "animation header-button",
                  "to": "/slideInX/properties/howto/keyframes",
                },
                "ref": null,
                "rendered": "animation properties",
                "type": [Function],
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/slideInX/properties/howto/general"
                />,
                <Route
                  component={[Function]}
                  path="/slideInX/properties/howto/properties"
                />,
                <Route
                  component={[Function]}
                  path="/slideInX/properties/howto/keyframes"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/slideInX/properties/howto/general",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/slideInX/properties/howto/properties",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/slideInX/properties/howto/keyframes",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Go!",
              "className": "close-info-popup-btn",
              "replace": false,
              "to": "/slideInX/properties",
            },
            "ref": null,
            "rendered": "Go!",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
